server/src/project/geometry/geometry.service.ts

```diff
@@ -283,8 +283,11 @@ export class GeometryService {
       .map(block => `${block.block.substring(1)}`);
   }
 
+  // Warning! Returns either null or an Object
   async getBblsGeometry(bbls = []) {
-    if (bbls === null || bbls.length === 0) return null;
+    // Sometimes Dynamics populates the BBLs array with one element: null.
+    // The carto fetch fails if the one element is null.
+    if (bbls === null || bbls.length === 0 || bbls[0] === null) return null;
```

Comment:
Thanks @godfreyyeung — I added a filter to the backend query to make sure it only returns not-null bbl.

My only problem with this approach is that a null value may be present in other parts of the array, at index 1 or 2, for example. Actually, one thing that'd help this is to just move line 292 above so that it filters out anything that isn't truthy (like nulls). Your call.

Reply:
Ahh, good suggestion. Will do!